gitpod-io/gitpod
# [server] Add generic tracing for API calls

## Description
We have had tracing in `server` for quite some time now. But it was never but in a central place in the API, but always felt like "bolt-on", which showed in noisy patterns like this:
```
const span = TraceContext.startSpan("someMethod", ctx);
try {
    // actual method implementation
} catch (e) {
    TraceContext.logError({ span }, e);
    throw e;
} finally {
    span.finish();
}
```

This PR moves this into a [central place](https://github.com/gitpod-io/gitpod/commit/decd846ab4542470bb71ee479992142612e6a40e#diff-cad2793f0b559ff7502b9ec959208584f789b9e5df185b3ed5d425c5c3b5d9edR312-R338), resulting in:
 a) all API calls being traced instead only a few
 b) the code being more readable


Open questions:
 - [x] Do we want to generically add _all parameters_ to every trace? ([example](https://github.com/gitpod-io/gitpod/compare/gpl/api-telemetry?expand=1#diff-d6fdb9bb15400f4fbba00db81b2ce6cb7f7bb566dd97b93b9fd4e3fc9a79f479R421-R423))
       Pro: even less, cleaner code
       Con: big parameters might blow up our storage/tracing infra
       @mads-hartmann I'd be super keen on your opinion here! :slightly_smiling_face: 



## Related Issue(s)

Fixes #

## How to test
 1. start a workspace, work with it, stop it
 2. check server logs for no errors (ignore `Error: getaddrinfo ENOTFOUND otel-collector`) :heavy_check_mark: 

## Release Notes

```release-note
NONE
```

## Documentation
